lib/exabgp/configuration/neighbor/__init__.py

```diff
@@ -239,9 +246,9 @@ def post (self):
 		if neighbor.range_size > 1 and not neighbor.passive:
 			return self.error.set('can only use ip ranges for the peer address with passive neighbors')
 
-		if neighbor.peer_address.top() in self._neighbors:
-			return self.error.set('duplicate peer definition %s' % neighbor.peer_address.top())
-		self._neighbors.append(neighbor.peer_address.top())
+		if neighbor.name() in self._neighbors:
+			return self.error.set('duplicate peer definition %s' % neighbor.name())
+		self._neighbors.append(neighbor.name())
```

Comment:
That's likely to break code - I would expect some code to rely on the IP address been here.

Reply:
I didn't find anything using the key of `self.neighbors` for anything else than iterating over it. And tests didn't trigger an error.